### PR TITLE
Master

### PR DESCRIPTION
**Pull request by leonfagan71**
Added {message} to all command actions and alert

### CODE REVIEW

**Comment by zedwick** [comment]
There's a lot of code repetition going on there. I had considered requesting you add the same to the other commands, but I couldn't think of a use case for the player sending a command which included their original message, for example. I decided it wasn't really needed.

However, if you'd like to make the placeholders available universally in all actions, we should have the logic in a single place so we don't have to maintain all these different lines of code to do the same thing.

**Comment by leonfagan71** [comment]
I agree, I'll probably make a function for it in a moment I'm just working
on a {match} which states the first word that is matched.

**Comment by leonfagan71** [comment]
Made it a function, simpler to change things.
{match} = word that matched to cause issue.
{message} = will display the message or command used
{player} = will display the playername

Example:
message: "{player}, you said '{match}' when saying {message}"

**Review by zedwick — CHANGES_REQUESTED** [review]
You seem to have duplicated config.yml and plugin.yml into the root, could you remove those please.

**Comment by leonfagan71** [comment]
I've removed plugin.yml and config.yml from the root, 
plugin.yml does cause an issue for me and requires it to be there, I've updated .gitignore for this to not be noticed in the future.

**Comment by zedwick** [comment]
Which IDE do you use? If you can open the project as a maven project, the pom.xml should provide the proper location for the plugin.yml and config.yml.

I think I'd prefer it if you didn't add little workarounds like that to the .gitignore. You can always add it to your local .gitignore if you want to still do that.

**Comment by leonfagan71** [comment]
Eclipse Java EE IDE.
For some reason it's ignoring the pom.xml file or it does not specify it in
it.
I will attempt to remove the global .gitignore and just use it locally.

Regards.

**Comment by leonfagan71** [comment]
Ahh, It looks like I was exporting instead of using Maven,
Issues resolved now.

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,5 @@ target/
 .BungeeChatFilter.iml
 bin/
 .settings/
-.git/
\ No newline at end of file
+.git/
+plugin.yml
```
Comment (zedwick) [inline .gitignore]: Remove plugin.yml line

File: src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java [hunk 2]
```diff
@@ -120,4 +132,23 @@ public boolean needsPermission(){
     public String getPermission() {
         return permission;
     }
+    public String findRWord(String message, String regex) throws java.lang.IllegalStateException{
+    	Pattern pattern = Pattern.compile(regex);
+    	Matcher m = pattern.matcher(message);
+    	m.find();
+    	return m.group(1) ;
```
Comment (zedwick) [inline src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java]: remove space here

File: src/main/resources/plugin.yml [hunk 3]
```diff
@@ -1,5 +0,0 @@
-name: BungeeChatFilter
-main : com.minecraftdimensions.bungeechatfilter.Main
-version: ${project.version}b-${build.number}
-author: Bloodsplat, Zedwick
-website: https://www.spigotmc.org/resources/bungeechatfilter.20596/
```
Comment (zedwick) [inline src/main/resources/plugin.yml]: Don't remove plugin.yml file.

File: src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java [hunk 4]
```diff
@@ -53,6 +51,9 @@ public boolean doesMessageContainRegex( String message ) {
 
     public void performActions( ChatEvent event, ProxiedPlayer player ) {
         String message = event.getMessage();
+        //get message word that matched.
+        
+
```
Comment (zedwick) [inline src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java]: Remove unused comment/line

File: src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java [hunk 5]
```diff
@@ -63,21 +64,32 @@ public void performActions( ChatEvent event, ProxiedPlayer player ) {
...
+            	String rp=customParams(actions.get( action )[0], player, event);
+				
+                player.disconnect( new TextComponent( TextComponent.fromLegacyText( Main.color( rp ) ) ) );
             } else if ( action.equals( "alert" ) ) {
-                String alert =   actions.get( action )[0].replace( "{player}", player.getDisplayName() );
+                
+                String rp=customParams(actions.get( action )[0], player, event);
                 if(message.split( " ", 2 ).length>1){
-                       alert =alert.replace("{arguments}", message.split( " ", 2 )[1] )    ;
+                	rp =rp.replace("{arguments}", message.split( " ", 2 )[1] )    ;
```
Comment (zedwick) [inline src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java]: I know it was there before, but could you remove the whitespace before the `;`

**Review by zedwick — CHANGES_REQUESTED** [review]
Got some changes for you to make before I can accept this, but once they're done should be ready.

**Comment by leonfagan71** [comment]
Have made those changes, any more to recommend?

**Review by zedwick — APPROVED** [review]

**Review by zedwick — CHANGES_REQUESTED** [review]
Could you please undo the changes to the whitespace at the beginning of all lines in Rule.java. The rest of the project uses spaces, where-as you have switched the lines in this file to tabs.

**Comment by leonfagan71** [comment]
Like that? 
all tabs replaced with spaces.
chucked it through a few syntax fixers to sort indentations and remove whitespace, that's why tabs existed.

**Comment by zedwick** [comment]
Right, I see.
Now I can see that a lot of the other original whitespace has been removed, the project has spaces inserted for readability (I assumed) throughout. I'm not necessarily completely opposed to that changing, but it shouldn't be part of this PR and should be applied throughout the project.

It also makes it difficult to easily see which lines of code has been changed, as I am seeing a lot of other unrelated lines changed due to the spaces being removed.

**Comment by leonfagan71** [comment]
Okay, I understand, 
I'm fairly new to GitHub in terms of actually adding to repositories,
I'll see if I can go back a bit.

**Comment by zedwick** [comment]
No worries.

**Pull request closed without merging**